service/src/main/java/com/commercetools/pspadapter/payone/KlarnaStartSessionHandler.java

```diff
@@ -114,33 +116,29 @@ private PayoneResult startSession(@Nonnull final PaymentWithCartLike paymentWith
                 StringUtils.isNotBlank(paymentWithCartLike.getPayment().getCustom().getFieldAsString(CLIENT_TOKEN))) {
             return new PayoneResult(OK_200, paymentWithCartLike.getPayment().getCustom().getFieldAsString(START_SESSION_RESPONSE));
         }
-        KlarnaRequestFactory requestFactory = new KlarnaRequestFactory(tenantConfig, new PayoneKlarnaCountryToLanguageMapper());
-        StartSessionRequestWithCart startSessionRequest = requestFactory.createStartSessionRequest(paymentWithCartLike);
 
-        List<UpdateAction<Payment>> updateActions = new ArrayList<UpdateAction<Payment>>();
-        final Map<String, String> response;
+        StartSessionRequestWithCart startSessionRequest = klarnaRequestFactory.createStartSessionRequest(paymentWithCartLike);
+
+
+        Map<String, String> response = null;
         try {
             response = payonePostService.executePost(startSessionRequest);
+            this.updatePayment(startSessionRequest, response, paymentWithCartLike);
         } catch (PayoneException paymentException) {
+            this.updatePayment(startSessionRequest, response, paymentWithCartLike);
             final String errorMessage = format("The 'startSession' Request to Payone failed for commercetools Payment" +
                     " with id '%s'.", paymentWithCartLike.getPayment().getId());
             logger.error(errorMessage, paymentException);
             return new PayoneResult(BAD_REQUEST_400, errorMessage);
         }
-
-        String responseBody = SphereJsonUtils.toJsonString(response);
-        updateActions.add(SetCustomField.ofObject(START_SESSION_RESPONSE, responseBody));
-
+        String responseBody = toJsonString(response);
```

Comment:
I would suggest to rename it `responseStr` for JSON string representation. Otherwise I thought `responseBody` represents a field called `body` inside `response`

Reply:
f88518648